src/types.rs

```diff
@@ -347,12 +349,68 @@ pub enum PlayerClass {
     Void,
 }
 
+#[derive(Debug, Clone)]
+pub struct Position {
+    x: f32,
+    y: f32,
+    z: f32,
+}
+
+impl<'de> Deserialize<'de> for Position {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        struct PositonVisitor {}
+
+        impl<'de> Visitor<'de> for PositonVisitor {
+            type Value = Position;
+
+            fn expecting(&self, formatter: &mut std::fmt::Formatter) -> std::fmt::Result {
+                formatter.write_str("a 3 element list of f32 seperated by space")
+            }
+
+            fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                let mut coords = [0f32; 3];
+                let splits = s.split(" ").collect::<Vec<&str>>();
+                if splits.len() != 3 {
+                    return Err(serde::de::Error::invalid_length(splits.len(), &self));
+                }
+
+                for (i, s) in splits.iter().enumerate() {
+                    coords[i] = s.parse().map_err(|_| serde::de::Error::invalid_type(serde::de::Unexpected::Str(s), &self))?;
+                }
+                let [x, y, z] = coords;
+                Ok(Position { x, y, z })
+            }
+
+            fn visit_string<E>(self, s: String) -> Result<Self::Value, E>
+            where
+                E: serde::de::Error,
+            {
+                self.visit_str(&s)
+            }
```

Comment:
```suggestion
```
This is not needed, as this is already the default implementation of `visit_string` (see [here](https://docs.serde.rs/src/serde/de/mod.rs.html#1509-1514)).